subsys/bluetooth/host/att.c

```diff
@@ -189,11 +189,18 @@ static int chan_send(struct bt_att_chan *chan, struct net_buf *buf,
 
 	chan->sent = cb ? cb : chan_cb(buf);
 
+	/* bt_l2cap_send_cb takes onwership of the buffer so take another
+	 * reference to restore the state in case an error is returned.
+	 */
+	net_buf_ref(buf);
+
 	err = bt_l2cap_send_cb(chan->att->conn, BT_L2CAP_CID_ATT,
 			       buf, att_cb(chan->sent),
 			       &chan->chan.chan);
 	if (err) {
 		net_buf_simple_restore(&buf->b, &state);
+	} else {
+		net_bug_unref(buf);
```

Comment:
I think you are introducing a bug here ;)

Reply:
It is releasing the reference we took to prevent hte buffer to be released while in bt_l2cap_send_cb, so what bug are you talking about here, you don't need to keep it a secrete.

Reply:
You have a spelling mistake, that's the "bug"